File: arm-fragment/core/src/main/java/com/iur/arm/fragment/core/anim/DefaultNoAnimator.java
package com.iur.arm.fragment.core.anim;


import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.NonNull;

public class DefaultNoAnimator extends FragmentAnimator implements Parcelable {
    public DefaultNoAnimator() {
        enter = 0;
        exit = 0;
        popEnter = 0;
        popExit = 0;
    }

    protected DefaultNoAnimator(Parcel in) {
        super(in);
    }

    @Override
    public void writeToParcel(@NonNull Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<DefaultNoAnimator> CREATOR = new Creator<>() {
        @Override
        public DefaultNoAnimator createFromParcel(Parcel in) {
            return new DefaultNoAnimator(in);
        }

        @Override
        public DefaultNoAnimator[] newArray(int size) {
            return new DefaultNoAnimator[size];
        }
    };
}

File: arm-fragment/core/src/main/java/com/iur/arm/fragment/core/anim/DefaultNoAnimator.kt
package com.iur.arm.fragment.core.anim

import android.os.Parcel
import android.os.Parcelable

class DefaultNoAnimator() : FragmentAnimator() {
    init {
        enter = 0
        exit = 0
        popEnter = 0
        popExit = 0
    }

    constructor(parcel: Parcel) : this() {
        enter = parcel.readInt()
        exit = parcel.readInt()
        popEnter = parcel.readInt()
        popExit = parcel.readInt()
    }

    companion object CREATOR : Parcelable.Creator<DefaultNoAnimator> {
        override fun createFromParcel(parcel: Parcel): DefaultNoAnimator = DefaultNoAnimator(parcel)

        override fun newArray(size: Int): Array<DefaultNoAnimator?> = arrayOfNulls(size)
    }
}
